apps/api/src/tasks/task.entity.ts

```diff
@@ -62,6 +63,9 @@ export class TaskEntity implements Task {
   @Column({ type: 'enum', enum: TaskPriority, default: TaskPriority.Longest })
   priority: TaskPriority;
 
+  @Column({ default: 0 })
+  priorityIndex: number;
```

Comment:
using FIFO is conflicting with this feature. could you explain the need exactly ?

Reply:
while supporting dependency first strategy, FIFO and other existing strategies becomes useless, but by introducing priority grouping, we can group tasks in groups by priority index and sort tasks inside each priority group using existing strategies like FIFO.